util/gutil/gutil_dump.go

```diff
@@ -110,9 +110,12 @@ func doDump(value interface{}, indent string, buffer *bytes.Buffer, option doDum
 	}
 	var (
 		reflectKind     = reflectValue.Kind()
-		reflectTypeName = reflectValue.Type().String()
+		reflectTypeName = "invalid"
```

Comment:
It may check the reflectValue at line 107:
```go
// Double check nil value.
if value == nil || reflectValue.IsNil() || !reflectValue.IsValid() {
	buffer.WriteString(`<nil>`)
	return
}
```

Reply:
I've tried your way of checking the reflectValue, and it brings the same mistake.
Not available to check reflectValue with IsNil().
The method I submitted is available.

Reply:
Yes,buffer will go to get infinitely bigger,then get OOM panic

Reply:
> Yes,buffer will go to get infinitely bigger,then get OOM panic
 
Any question carrying on this PR?